### DISCUSSION TITLE
Hierarchy in packages nugets .targets

### DISCUSSION THREAD

**Discussion by jmanriquecruz** [NuGet/Home#12478] [Q&A] [answered]
Hello guys, I am building a parent nuget package that is an SDK and that uses a third-party library that is also in a nuget "framework.nupkg", my application references the nuget of the sdk, both packages use a .targets file but when I include the reference to my project the child package does nothing and the child .targets does not appear in the references

![image](https://user-images.githubusercontent.com/61441532/224167562-9435d203-36ac-4ba8-8abd-e1958026c0cb.png)

in the application the .targets of the sdk is copied but the one of the child nuget is not there and as a consequence it does nothing
![image](https://user-images.githubusercontent.com/61441532/224167346-d9da25e8-3902-4c83-b328-49b5132ca023.png)

I appreciate any help to help me solve the problem
 the first library nuget uses .targets because it has libraries from various operating systems

**Comment by TheXenocide** [comment, marked as answer]
By default, props and targets from transitive packages are not automatically included in downstream consumers. If you want a dependency of a dependency to load props/targets you can add copies of them to the `buildTransitive` package folder (in addition to the `build` folder) which is automatically included. Alternatively (and this is more for conditions where the transitive dependency isn't in your control and only includes props/targets under the `build` package folder) you can add `IncludeAssets="build"` to your `PackageReference` to have its generated nuspec indicate to consuming libraries that they should also include build props/targets for its declared dependency.

As a side note, there may be some naming confusion with you [Redacted].SDK package now that custom project SDKs are becoming more common (e.g. `<Project Sdk="My.Custom.Sdk">` instead of `<Project Sdk="Microsoft.NET.Sdk">` for the first line in your csproj). These custom SDKs are also nuget packages, but wouldn't use transitive dependencies the way you define them here and have some different naming conventions for auto-included props/targets files, etc.

**Comment by jmanriquecruz** [comment]
Hello @TheXenocide, thanks for your help, although the last thing you told me was not very clear, I tell you that what I want is for the third-party library that I am using to be a separate nuget and the sdk for the implementation of those dlls to be another nuget that's why I have the dependency, because of the requirement.

**Reply by TheXenocide** [reply]
If you control the package authoring of [Redacted]Framework.Multiplatform (which would make it first-party) then pack your props/targets files to the `buildTransitive` package directory (in addition to the `build` package directory). This indicates that the build-time behavior is intended to be added to projects, even if they don't *directly* reference the [Redacted]Framework.Multiplatform package (normally `build` directory behavior only applies to projects that directly reference the package). 

If you don't control [Redacted]Framework.Multiplatform (making it third-party) then in your [Redacted].Sdk project, where you have `<PackageReference Include="[Redacted]Framework.Multiplatform" Version="#.#.#" />` make that `<PackageReference Include="[Redacted]Framework.Multiplatform" Version="#.#.#" IncludeAssets="build" />`  so that when a project references your SDK package, it knows that it also needs to include behavior from the `build` package directory.

**Reply by jmanriquecruz** [reply]
Thank you so much bro, you saved my day.
